2.ServiceTracker_App/App/data-api/app.js

```diff
@@ -3,65 +3,60 @@ const createError = require('http-errors');
 const dayjs = require('dayjs');
 const express = require('express');
 const logger = require('morgan');
-const mongoose = require('mongoose');
 const path = require('path');
 const relativeTime = require('dayjs/plugin/relativeTime');
 
 dayjs.extend(relativeTime);
-//connectTimeoutMS
 global.start = dayjs().valueOf();
 
 if (process.env.NODE_ENV != 'container') {
   require('dotenv').config({ path: path.join(__dirname, '.env.local') });
 }
 
-mongoose.set('useCreateIndex', true);
-mongoose.set('useFindAndModify', false);
-
-require('./models/mongo/flights');
-require('./models/mongo/latestFlight');
-require('./models/mongo/quakes');
-require('./models/mongo/latestQuake');
-require('./models/mongo/weather');
-require('./models/mongo/latestWeather');
+const flights = require('./models/database/flights');
+const latestFlight = require('./models/database/latestFlight');
+const quakes = require('./models/database/quakes');
+const latestQuake = require('./models/database/latestQuake');
+const weather = require('./models/database/weather');
+const latestWeather = require('./models/database/latestWeather');
+
+const Sequelize = require('sequelize');
+const sequelize = new Sequelize(process.env.DATABASE_NAME, process.env.DATABASE_USER, process.env.DATABASE_PASSWORD, {
```

Comment:
Does `sequelize` create the database if it does not exist? It looks like when running postgres in-cluster a database name can be specified (https://github.com/oam-dev/samples/pull/28/files#diff-8e1a64899df56ff23de0b22c0bfa1b80R33). This is unfortunately not always true with managed Postgres offerings. We may need to build in some logic to create the database if not exist, or use an `initContainer` to do so.

Reply:
Which managed postgres will we use to demo? I would like to make this sample in 2 scenarios: reproducible locally and in the demo environment.

Reply:
And, yes. sequelize (love that name) created the database too.

Reply:
I validated this with Azure. You are right, the database must exist because the cloud provider does not give permission to create database via SQL connection, only via portal. Then, sequelize tries to create and crashes. If the database already exists, then it creates the tables.

For postgres, Azure gives a single pre-created database called `postgres`. These are the settings that worked to use Postgres in Azure (ask me for the removed values if you want to try this out too):
```
        - name: dbuser
          value: "REMOVED"
        - name: dbpassword
          value: "REMOVED"
        - name: dbhostname
          value: "REMOVED.postgres.database.azure.com"
        - name: dbport
          value: "5432"
        - name: dbname
          value: "postgres"  
        - name: dbdriver
          value: "postgres"
```

Now, for MSSQL. I could connect but the current model does not work because MSSQL has no support for JSON data type (JSON is just text in MSSQL). Sequelize doc says that JSON type is not supported in MSSQL.
```
        - name: dbuser
          value: "DELETED"
        - name: dbpassword
          value: "DELETED"
        - name: dbhostname
          value: "DELETED.database.windows.net"
        - name: dbport
          value: "1433"
        - name: dbname
          value: "hackfest"
        - name: dbdriver
          value: "mssql"
        - name: dboptions
          value: "{ \"options\": { \"encrypt\": true } }"
```

Reply:
Thanks for looking into this! We can adjust how these values are being passed to the application once we have a working runtime implementation 👍